File: 724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int totalSum = 0;
        for (int x : nums) totalSum += x;

        int leftSum = 0;

        for (int i = 0; i < nums.size(); i++) {
            int rightSum = totalSum - leftSum - nums[i];

            if (leftSum == rightSum) {
                return i;
            }

            leftSum += nums[i];
        }
        return -1;
    }
};

File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left=0
        sum=0
        for i in range(len(nums)):
            sum+=nums[i]
        if sum-nums[0]==0:
            return 0  
        for i in range(1,len(nums)):
            left+=nums[i-1]
            right=sum-left-nums[i]
            if left==right:
                return i
        return -1